app/src/command.rs

```diff
@@ -28,18 +29,27 @@ use {
 
 impl<'app, D: Dependencies<'app>> Application<'app, D> {
     #[tracing::instrument(level = "debug", skip(self, attributes))]
-    pub fn start_block_build(&mut self, attributes: PayloadForExecution, id: PayloadId) {
-        if self
+    pub(crate) fn start_block_build(
+        &mut self,
+        attributes: PayloadForExecution,
+        id: PayloadId,
+    ) -> Result<(), UnrecoverableAppFailure> {
+        let payload_exists = self
             .payload_queries
             .by_id(&self.storage_reader, id)
-            .unwrap()
-            .is_some()
-        {
-            return;
+            .map_err(|e| {
+                tracing::error!(
+                    "Failure during `start_block_build`. Payload queries failed: {e:?}"
+                );
```

Comment:
Shouldn't the above `#[instrument]` macro 
already take care of the scoping? AFAICT it names the span identically as the function within which an event was created.